### PR TITLE
Purchase Delivery Module for Odoo v8

### PR DESCRIPTION
**Pull request by jpoa**
Module now works with Odoo 8. Removed the last view for picking as it seems to be already present (at least on my install), if later I see its needed it can be readded. At least this way some of the needed changes are already documented.

### CODE REVIEW

File: purchase_delivery/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+##############################################################################
+
+
+{
+    'name': 'Purchase Delivery Costs',
+    'version': '1.0',
+    'category': 'Purchase Management',
+    'description': """
+Allows you to add delivery methods in purchase orders and picking.
+==============================================================
```
Comment (pedrobaeza) [inline purchase_delivery/__openerp__.py:29]: Put the same length of the title

File: purchase_delivery/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+#
+##############################################################################
+
+
+{
+    'name': 'Purchase Delivery Costs',
+    'version': '1.0',
+    'category': 'Purchase Management',
+    'description': """
+Allows you to add delivery methods in purchase orders and picking.
```
Comment (pedrobaeza) [inline purchase_delivery/__openerp__.py:28]: Remove the dot

Comment (pedrobaeza) [inline purchase_delivery/__openerp__.py:28]: Extract this into a README.rst file

File: purchase_delivery/model/delivery.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                    continue
+                if grid.src_zip_from and (src.zip or '') < grid.src_zip_from:
+                    continue
+                if grid.src_zip_to and (src.zip or '') > grid.src_zip_to:
+                    continue
+                return grid.id
+        return False
+
+
+class delivery_grid(orm.Model):
```
Comment (pedrobaeza) [inline purchase_delivery/model/delivery.py:75]: Use CamelCase for class names

File: purchase_delivery/model/purchase.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        line_obj = self.pool.get('purchase.order.line')
+        grid_obj = self.pool.get('delivery.grid')
+        carrier_obj = self.pool.get('delivery.carrier')
+        acc_fp_obj = self.pool.get('account.fiscal.position')
+        for order in self.browse(cr, uid, ids, context=context):
+            src_address_id = order.partner_id
+            dest_address_id = order.dest_address_id or False
+            if (
+                not dest_address_id
+                and order.warehouse_id
```
Comment (gmathers) [inline purchase_delivery/model/purchase.py:61]: I'm getting this error trying to click the "Add in Quote" button. The warehouse_id field is changed in V8

 File "/home/systemadmin/PycharmProjects/iii-addons/purchase_delivery/model/purchase.py", line 61, in delivery_set
    and order.warehouse_id
AttributeError: 'purchase.order' object has no attribute 'warehouse_id'

**Comment by yelizariev** [comment]
I've made some updates for this module. Work is not finished, but "Add in Quote" button works.

https://github.com/yelizariev/eficent-odoo-addons/tree/8.0/purchase_delivery

cc: @jbeficent

**Comment by jgrandguillaume** [comment]
Hi,

I've made some tests. Trying to generate the line from delivery method in PO give me this error:

> File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3113027-157-342f15/openerp/addons/purchase_delivery/model/purchase.py", line 61, in delivery_set
>    and order.warehouse_id
> AttributeError: 'purchase.order' object has no attribute 'warehouse_id'

Regards,

Joël

**Comment by jpoa** [comment]
Hi there,

I am willing to work a bit on this in the coming weeks, seems to be some issue with fields that changed location/name.

Cheers!

On 14 Aug 2015 at 15:57:25, Joël Grand-Guillaume (notifications@github.com) wrote:

Hi,

I've made some tests. Trying to generate the line from delivery method in PO give me this error:

File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3113027-157-342f15/openerp/addons/purchase_delivery/model/purchase.py", line 61, in delivery_set
and order.warehouse_id
AttributeError: 'purchase.order' object has no attribute 'warehouse_id'

Regards,

Joël

**Comment by yvaucher** [comment]
@jpoa I added a WIP tag. When you are done just ping us so we remove it.

**Comment by yelizariev** [comment]
@jpoa don't forget to take a look on my commits. I've fixed issue with warehouse_id mentioned above.

Also, I suggest to work on fork on eficent-odoo-addons, because the module is originally from there.

**Pull request closed without merging**